### PR TITLE
Propagate the context to all functions

### PR DESCRIPTION
**Pull request by qbarrand**
Use the controller's `context.Context` in all functions that need it, instead of `context.TODO()`.

### CODE REVIEW

**Comment by qbarrand** [comment]
/uncc @bthurber 
/cc @ybettan @pacevedom
PTAL. This PR should not modify any business logic, just a chore that makes the code more consistent.

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -141,7 +141,8 @@ func (p *proxy) setupContainersProxy(containers []interface{}) error {
 	return nil
 }
 
-func (p *proxy) ClusterConfiguration() (Configuration, error) {
+func (p *proxy) ClusterConfiguration(ctx context.Context) (Configuration, error) {
+	//>>>>>>> 53b720e7 (Propagate the context to all functions.)
```
Comment (pacevedom) [inline pkg/proxy/proxy.go]: Leftover from a conflict?

File: test/deploy/deploy.go [hunk 2]
```diff
@@ -45,7 +46,7 @@ func main() {
 	manifests := assetsInterface.GetFrom(*path)
 
 	for _, manifest := range manifests {
-		if err = framework.CreateFromYAML(manifest.Content, cl); err != nil {
+		if err = framework.CreateFromYAML(context.Background(), manifest.Content, cl); err != nil {
```
Comment (pacevedom) [inline test/deploy/deploy.go]: Using `TODO()` in tests and filter package, should we do the same here? `Background()` is functionally equivalent.

File: test/undeploy/undeploy.go [hunk 3]
```diff
@@ -41,7 +42,9 @@ func main() {
 		log.Fatalf("Error getting a controller client: %v", err)
 	}
 
-	if err = framework.DeleteAllSpecialResources(cl); err != nil {
+	ctx := context.Background()
```
Comment (pacevedom) [inline test/undeploy/undeploy.go]: `TODO()` instead?

Comment (qbarrand) [reply test/undeploy/undeploy.go]: Yes, let us.

File: test/deploy/deploy.go (hunk 2 above)
Comment (qbarrand) [reply test/deploy/deploy.go]: Yes, let us.

File: pkg/proxy/proxy.go (hunk 1 above)
Comment (qbarrand) [reply pkg/proxy/proxy.go]: Oops.

**Review by pacevedom — APPROVED** [review]

**Pull request merged**